project_key/models/project_project.py

```diff
@@ -30,6 +30,14 @@ def _onchange_project_name(self):
             else:
                 rec.key = ""
 
+    @api.depends("key", "name")
+    def _compute_display_name(self):
+        for project in self:
+            if project.key:
+                project.display_name = f"[{project.key}] {project.name}"
+            else:
+                project.display_name = f"{project.name}"
```

Comment:
Can you please check if the following suggestion is okay? This is to avoid conflicts with other modules that manipulate with the display name.
```suggestion
        super()._compute_display_name()
        for project in self:
            if project.key:
                project.display_name = f"[{project.key}] {project.display_name}"
        return
```

The same applies to the method for `project.task`.

Reply:
Good catch. I have adapted the code based on your suggestion. Thanks 👍